provider/slof.py

```diff
@@ -0,0 +1,194 @@
+"""
+Module for providing common interface to test SLOF component.
+"""
+
+import re
+import time
+import logging
+import os
+
+from virttest import utils_misc
+
+
+class LazyReader(object):
+    """
+    Custom a reading file container(iterable).
+    """
+    def __init__(self, vm):
+        self._filename = vm.serial_console_log
+        self._fd = open(self._filename)
+
+    def __iter__(self):
+        pos = 0
+        with open(self._filename) as f:
+            for line in f:
+                yield line, pos
+                pos += 1
+
+    def __len__(self):
```

Comment:
Is this __len__ only  used in `line 39` : `start, stop, step = index.indices(len(self))`?
if you run
 ```python
lr = LazyReader(vm)
print len(lr)   # file line number
print len(lr)   # 0
print len(lr)   # 0
```

Reply:
Thanks for your comments, have replaced it with the following protected function:
```
    def _get_len(self):
        num = 0
        while 1:
            try:
                next(self._fd)
            except StopIteration:
                return num
            num += 1
```